drivers/flash/flash_stm32_ospi.c

```diff
@@ -367,7 +365,6 @@ static int stm32_ospi_write_enable(OSPI_HandleTypeDef *hospi, uint8_t nor_mode,
 		s_command.Instruction = SPI_NOR_OCMD_WREN;
 	} else {
 		s_command.Instruction = SPI_NOR_CMD_WREN;
-		s_command.InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE;
```

Comment:
Many dual and quad mode ICs use 1 line mode for that.
By removing

> `s_command.InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE;`

the mode is enforced to be 2 or 4 lines in the prepare cmd while it shall be 1 line.

Reply:
rcing the `s_command.InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE` for all non-ospi modes
after the prepare command which sets the s_command.DataMode